crates/beerus-rpc/src/lib.rs

```diff
@@ -102,12 +101,9 @@ impl BeerusApiServer for BeerusRpc {
             .map_err(|_| Error::from(BeerusApiError::TxnHashNotFound))
     }
 
-    async fn get_block_transaction_count(
-        &self,
-        block_id_type: String,
-        block_id: String,
-    ) -> Result<u64, Error> {
-        let block_id = block_id_string_to_block_id_type(&block_id_type, &block_id).unwrap();
+    async fn get_block_transaction_count(&self, block_id: String) -> Result<u64, Error> {
+        let block_id: BlockId = serde_json::from_str(&block_id)
+            .map_err(|e| Error::Call(CallError::InvalidParams(anyhow::anyhow!(e.to_string()))))?;
```

Comment:
Valid for all other instances: 

Shouldn't we use BeerusApiError error handling here?

As follows: .map_err(|_| Error::from(BeerusApiError::InvalidCalldata))

As we are handling errors this way on other endpoints,

Please explain

Reply:
Oh yeah, that's my bad. I'll change it to use BeerusApiError.